server/src/Masna3/Server/Error.hs

```diff
@@ -1,12 +1,54 @@
 module Masna3.Server.Error where
 
 import Data.Aeson
+import GHC.Generics (Generic)
+import Masna3.Api.File.FileId
 import Servant (ServerError (..), err500)
 
 data Masna3Error
   = TooManyRows Text
+  | FileNotFound FileId
+  | InvalidTransition InvalidTransitionError
   deriving stock (Eq, Ord, Show)
 
+data InvalidTransitionError
+  = NotPendingToUploaded FileId
+  deriving stock (Eq, Generic, Ord, Show)
+  deriving anyclass (ToJSON)
+
 toServerError :: Masna3Error -> ServerError
 toServerError = \case
-  TooManyRows t -> err500{errBody = encode t}
+  TooManyRows t ->
+    err500
+      { errBody =
+          encode
+            ErrorResponse
+              { errorType = "TooManyRows"
+              , context = Just $ object ["ownerId" .= t]
+              }
+      }
+  FileNotFound t ->
+    err500
+      { errBody =
+          encode
+            ErrorResponse
+              { errorType = "FileNotFound"
+              , context = Just $ object ["fileId" .= t]
+              }
+      }
```

Comment:
For "\<something\> not found" errors, we use the [404](https://hackage-content.haskell.org/package/servant-server-0.20.3.0/docs/Servant-Server.html#v:err404) status. You can find a list of the HTTP statuses that are available to us [here](https://http.cat/), although I suggest that we restrict ourselves to:

* 404 when a resource is not found
* 403 when authentication fails (it's not used yet)
* 500 when we have a server-side exception